directives/authDirective.js

```diff
@@ -0,0 +1,195 @@
+const { SchemaDirectiveVisitor } = require('apollo-server-express');
+const {
+  UnauthenticatedError,
+  UnauthorizedError,
+  NotFoundError,
+} = require('errors');
+const { defaultFieldResolver } = require('graphql');
+const requestContext = require('talawa-request-context');
+
+const Event = require('../models/Event');
+const Group = require('../models/Group');
+const GroupChat = require('../models/GroupChat');
+const MembershipRequest = require('../models/MembershipRequest');
+const Organization = require('../models/Organization');
+
+class AuthenticationDirective extends SchemaDirectiveVisitor {
+  visitFieldDefinition(field) {
+    const resolver = field.resolve || defaultFieldResolver;
+    field.resolve = async (root, args, context, info) => {
+      const { requires } = this.args;
+
+      if (context.expired || !context.isAuth)
+        throw new UnauthenticatedError(
+          requestContext.translate('user.notAuthenticated'),
+          'user.notAuthenticated',
+          'userAuthentication'
+        );
+
+      if (requires === 'Admin' && args.eventId) {
```

Comment:
@aniketbiswas21  can you pull out these blocks into helper functions.


if (requires === 'Admin' && args.xxx) {
      <block>
}

Reply:
@impiyush83 Can you elaborate sir? :sweat_smile: